src/sql/workbench/electron-browser/modelComponents/diffeditor.component.ts

```diff
@@ -79,8 +81,20 @@ export default class DiffEditorComponent extends ComponentBase implements ICompo
 		this.editorUriRight = uri2.toString();
 
 		let cancellationTokenSource = new CancellationTokenSource();
-		let editorinput1 = this._instantiationService.createInstance(UntitledEditorInput, uri1, false, 'plaintext', '', '');
-		let editorinput2 = this._instantiationService.createInstance(UntitledEditorInput, uri2, false, 'plaintext', '', '');
+		this._textModelService.registerTextModelContentProvider('diffEditor', {
+			provideTextContent: (resource: URI): Promise<ITextModel> => {
+				if (resource.scheme === 'diffEditor') {
+					let modelContent = '';
+					let languageSelection = this._modeService.create('plaintext');
+					return Promise.resolve(this._modelService.createModel(modelContent, languageSelection, resource));
+				}
+
+				return Promise.resolve(null!);
+			}
+		});
+
+		let editorinput1 = this._instantiationService.createInstance(ResourceEditorInput, 'source', 'description', uri1, undefined);
+		let editorinput2 = this._instantiationService.createInstance(ResourceEditorInput, 'target', 'description', uri2, undefined);
```

Comment:
Do  you need the description for anything? It looks like the constructor takes null as a value and generally I prefer to explicitly pass null instead of dummy values

Reply:
replaced with undefined